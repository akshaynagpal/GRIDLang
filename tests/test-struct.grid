Player good
{ 
int x; 
int y; 
string s; 
} 

int gameloop()
{ 
Player good p1; 
p1.x = 10;
p1.y = 20;
p1.s = "struct works";
print(p1.x);
print(p1.y);
print(p1.s);
return 0; 
}
