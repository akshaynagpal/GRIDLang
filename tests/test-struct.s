	.file	"tests/test-struct.ll"
	.text
	.globl	main
	.align	16, 0x90
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# BB#0:                                 # %entry
	movl	$10, -4(%rsp)
	movl	$20, -8(%rsp)
	movq	$.Lname, -16(%rsp)
	xorl	%eax, %eax
	ret
.Ltmp0:
	.size	main, .Ltmp0-main
	.cfi_endproc

	.type	.Lfmt,@object           # @fmt
	.section	.rodata.str1.1,"aMS",@progbits,1
.Lfmt:
	.asciz	"%d\n"
	.size	.Lfmt, 4

	.type	.Lmmt,@object           # @mmt
.Lmmt:
	.asciz	"%s\n"
	.size	.Lmmt, 4

	.type	.Lname,@object          # @name
.Lname:
	.asciz	"struct works"
	.size	.Lname, 13


	.section	".note.GNU-stack","",@progbits
