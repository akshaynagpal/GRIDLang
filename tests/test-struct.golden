10
20
struct works